### PR TITLE
SRVCOM-488: OpenShift Serverless Installation 

### PR DESCRIPTION
**Pull request by nebelsprachen**
- branch/enterprise-4.1 
- branch/enterprise-4.2

### CODE REVIEW

File: modules/serverless-cluster-sizing-requirements.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+= Cluster size requirements
+
+Determine how many nodes and pods you require for your OpenShift Container Platform cluster. Cluster scalability correlates to the number of pods in a cluster environment. That number influences the other numbers in your setup.
+
+// Scalability, planning requirements, and limits
+include:scalability_and_performance/planning-your-environment-according-to-object-limits.adoc[leveloffset=+1]
+
+Your cluster must be sized appropriately to ensure the instructions work properly. To do so, you can use the MachineSet API to manually scale your cluster up to the desired size.
+
+An Openshift cluster with 16 CPUs and 40 GB memory is the minimum requirement for getting started with your first serverless application. This usually means you will have to scale up one of the default MachineSets by two additional machines.
```
Comment (markusthoemmes) [inline modules/serverless-cluster-sizing-requirements.adoc]:
```suggestion
An Openshift cluster with 10 CPUs and 40 GB memory is the minimum requirement for getting started with your first serverless application. This usually means you will have to scale up one of the default MachineSets by two additional machines.
```

We corrected that in the doc already.

File: modules/serverless-cluster-sizing-requirements.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+[id="cluster-size-requirements_context"]
+= Cluster size requirements
+
+Determine how many nodes and pods you require for your OpenShift Container Platform cluster. Cluster scalability correlates to the number of pods in a cluster environment. That number influences the other numbers in your setup.
+
+// Scalability, planning requirements, and limits
+include:scalability_and_performance/planning-your-environment-according-to-object-limits.adoc[leveloffset=+1]
+
+Your cluster must be sized appropriately to ensure the instructions work properly. To do so, you can use the MachineSet API to manually scale your cluster up to the desired size.
+
```
Comment (markusthoemmes) [inline modules/serverless-cluster-sizing-requirements.adoc]: Add a note here:

> Note: All numbers given only relate to the pool of worker machines of the Openshift cluster. Master nodes are not used for general scheduling and thus are omitted.

Adjusted in the doc already.

File: modules/serverless-installing-knative-serving.adoc [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+[id="installing-knative-serving_{context}"]
+= Installing Knative Serving
+
+You need to create a `KnativeServing` object to install Knative Serving using the Openshift Serverless Operator.
+
+
+[IMPORTANT]
+====
+This sample YAML file is provided for reference.
```
Comment (markusthoemmes) [inline modules/serverless-installing-knative-serving.adoc]: What does this tell me as a user?

File: modules/serverless-installing-service-mesh-control-plane.adoc [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     enabled: false
+   prometheus:
+     enabled: false
+   grafana:
+     enabled: false
+   sidecarInjectorWebhook:
+     enabled: false
+----
+<1> Required. The installation program prompts you for this value.
+<2> Autoscaling is disabled in this version, as this is currently not intended for production use.
```
Comment (markusthoemmes) [inline modules/serverless-installing-service-mesh-control-plane.adoc]: Maybe slightly reword this to sound more like: This is a minimum version to kick the tires. Please adjust these values to your production needs as necessary?

File: modules/serverless-installing-service-mesh-control-plane.adoc [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+   tracing:
+     enabled: false
+   prometheus:
+     enabled: false
+   grafana:
+     enabled: false
+   sidecarInjectorWebhook:
+     enabled: false
+----
+<1> Required. The installation program prompts you for this value.
```
Comment (markusthoemmes) [inline modules/serverless-installing-service-mesh-control-plane.adoc]: Which installation program? This seems like a duplicate to the note below?

**Review by markusthoemmes — COMMENTED** [review]
Left a few comments, I think a preview is needed in this case to see what the includes result in.

File: modules/serverless-cluster-sizing-requirements.adoc [hunk 6]
```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+
+[id="cluster-size-requirements_context"]
+= Cluster size requirements
+
+Determine how many nodes and pods you require for your OpenShift Container Platform cluster. Cluster scalability correlates to the number of pods in a cluster environment. That number influences the other numbers in your setup.
```
Comment (abrennan89) [inline modules/serverless-cluster-sizing-requirements.adoc]: "You must determine"

Comment (abrennan89) [inline modules/serverless-cluster-sizing-requirements.adoc]: Ensure that {product-title} variable is being used instead of manually writing "OpenShift Container Platform" each time

Comment (abrennan89) [inline modules/serverless-cluster-sizing-requirements.adoc]: "Cluster scalability correlates to the number of pods in a cluster environment. That number influences the other numbers in your setup." I don't think it's clear enough what this means. Please look at rewriting this sentence so it explains better.

File: modules/serverless-cluster-sizing-requirements.adoc [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+// Scalability, planning requirements, and limits
+include:scalability_and_performance/planning-your-environment-according-to-object-limits.adoc[leveloffset=+1]
```
Comment (abrennan89) [inline modules/serverless-cluster-sizing-requirements.adoc]: It looks like this include is inside a module - shouldn't  modules/serverless-cluster-sizing-requirements.adoc  then be an assembly instead?

File: modules/serverless-cluster-sizing-requirements.adoc [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+Your cluster must be sized appropriately to ensure the instructions work properly. To do so, you can use the MachineSet API to manually scale your cluster up to the desired size.
```
Comment (abrennan89) [inline modules/serverless-cluster-sizing-requirements.adoc]: We should add some details about what MachineSet API is or a link to existing docs etc; to me it's confusing to just mention this but not have a procedure or some explanation?

File: modules/serverless-cluster-sizing-requirements.adoc [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+An Openshift cluster with 16 CPUs and 40 GB memory is the minimum requirement for getting started with your first serverless application. This usually means you will have to scale up one of the default MachineSets by two additional machines.
+
+For more advanced use-cases, such as using Openshift Logging, Monitoring, Metering and Tracing, you will need to deploy more resources. Recommended requirements for such use-cases are 24 vCPUs and 96GB of memory.
+
+// Add or remove an instance of a machine in a MachineSet
```
Comment (abrennan89) [inline modules/serverless-cluster-sizing-requirements.adoc]: There isn't enough content here about what machinesets are IMHO, maybe add a link to some context info or add something about how these work regarding serverless?

File: modules/serverless-installing-knative-serving.adoc (hunk 3 above)
Comment (abrennan89) [reply modules/serverless-installing-knative-serving.adoc]: Should probably be "This sample yaml shows the minimum configuration to create a KnativeServing object:" or something like that

File: modules/serverless-installing-knative-serving.adoc [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+
+.Procedure
+. Copy the sample YAML file into `serving.yaml` and apply it using:
++
+----
+oc apply -f serving.yaml
+----
++
+. Verify the installation is complete by using the command:
```
Comment (abrennan89) [inline modules/serverless-installing-knative-serving.adoc]: What should the output be to verify against?

File: modules/serverless-installing-service-mesh-control-plane.adoc [hunk 11]
```diff
@@ -0,0 +1,103 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+= Installing the ServiceMeshControlPlane
+
+After you install the ServiceMesh operator, you can install the control plane. The following procedure installs a version of ServiceMesh that acts as an ingress to your applications.
+
+[IMPORTANT]
+====
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh-control-plane.adoc]: again, explain exactly what the sample is for, there's no context

File: modules/serverless-installing-service-mesh-control-plane.adoc (hunk 5 above)
Comment (abrennan89) [reply modules/serverless-installing-service-mesh-control-plane.adoc]: This looks like it refers to the namespace. I'd remove the 'installation program' bit and just call out the fields that are required.
I'm not sure there's a need to include that whole huge code sample - does this not exist in service mesh docs or somewhere else already?

File: modules/serverless-installing-service-mesh-control-plane.adoc [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+
+====
+[NOTE]
+Running ServiceMesh with a sidecar injection enabled with Openshift Serverless is currently not recommended. 
+====
+
+====
+[NOTE] 
+You need to install the control plane into the `istio-system` namespace. Other namespaces are currently not supported.
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh-control-plane.adoc]: Move this note to earlier in the section

File: modules/serverless-installing-service-mesh-control-plane.adoc [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+====
+
+
+.Prerequisite
+* An account with cluster administrator access.
+* The ServiceMesh operator must be installed. 
+
+
+.Procedure
+. Copy the sample YAML file into a `smcp.yaml` file.
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh-control-plane.adoc]: what sample yaml file? please put it here instead so it makes sense

File: modules/serverless-installing-service-mesh-member-roll.adoc [hunk 14]
```diff
@@ -0,0 +1,45 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id="installing-service-mesh-member-roll_{context}"]
+= Installing a ServiceMeshMemberRoll
+
+For applications to make use of the deployed control plane and ingress, their namespaces need to be part of a member roll.
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh-member-roll.adoc]: "need to be added to a member roll"

File: modules/serverless-installing-service-mesh-member-roll.adoc [hunk 15]
```diff
@@ -0,0 +1,45 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id="installing-service-mesh-member-roll_{context}"]
+= Installing a ServiceMeshMemberRoll
+
+For applications to make use of the deployed control plane and ingress, their namespaces need to be part of a member roll. 
+
+.ServiceMeshMemberRoll Custom Resource Example
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh-member-roll.adoc]: again, no context, what is this example for, explain it

File: modules/serverless-installing-service-mesh-member-roll.adoc [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+A multi-tenant control plane installation only affects namespaces configured as part of the Service Mesh. You must specify the namespaces associated with the Service Mesh in a `ServiceMeshMemberRoll` resource located in the same namespace as the `ServiceMeshControlPlane` resource and name it `default`.
+
+
+.Prerequisites
+* An installed, verified Service Mesh Operator.
+* A custom resource file that defines the parameters of your Red Hat OpenShift Service Mesh control plane.
+
+
+.Procedure
+. Create a YAML file that replicates the ServiceMeshMemberRoll Custom Resource sample.
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh-member-roll.adoc]: show all the examples here with the actual procedure steps to give context, random examples with no steps don't have enough context

File: modules/serverless-installing-service-mesh.adoc [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+====
+
+== Installing ServiceMesh without Kiali and Jaeger Operators
+
+.Prerequisites
+* An account with cluster administrator access.
+
+
+.Procedure
+. Log into your OpenShift Container Platform installation as a cluster administrator.
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh.adoc]: {product-title}

File: modules/serverless-installing-service-mesh.adoc [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+. Run the following commands to install:
```
Comment (abrennan89) [inline modules/serverless-installing-service-mesh.adoc]: "Run the following commands to install Service Mesh:"

File: modules/serverless-installing-the-serverless-operator.adoc [hunk 19]
```diff
@@ -0,0 +1,10 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+= Installing the OpenShift Serverless Operator
+
+The OpenShift Serverless Operator can be installed by using the operator indexed in the OpenShift OperatorHub.
```
Comment (abrennan89) [inline modules/serverless-installing-the-serverless-operator.adoc]: "The OpenShift Serverless Operator can be installed by using the {product-title} instructions on installing an Operator."

Include CLI and web console modules?

**Review by abrennan89 — CHANGES_REQUESTED** [review]

File: modules/serverless-installing-service-mesh-control-plane.adoc (hunk 5 above)
Comment (markusthoemmes) [reply modules/serverless-installing-service-mesh-control-plane.adoc]: Not in this form, it's actually the recommended and guaranteed-to-work well way to install ServiceMesh for the serverless setting.

Comment (markusthoemmes) [reply modules/serverless-installing-service-mesh-control-plane.adoc]: We could however pick apart and call out the "important" bits (i.e. disabling the actual mesh injection)

File: modules/serverless-installing-service-mesh-member-roll.adoc (hunk 16 above)
Comment (joaedwar) [reply modules/serverless-installing-service-mesh-member-roll.adoc]: @abrennan89  I don't understand what you mean by this. Would you like me to include the YAML file in the procedure? Looking at OpenShift docs examples, it seems they have the YAML examples outside of the procedures

Comment (abrennan89) [reply modules/serverless-installing-service-mesh-member-roll.adoc]: If you're including the example outside the procedure, I'd make sure it's explained with some context about what individual fields are required for basically.
E.g. in the service mesh docs here https://docs.openshift.com/container-platform/4.1/service_mesh/service_mesh_install/installing-ossm.html there are tables which explain what the fields do.
I'm not sure we need to explain every field in a yaml example because some of it is probably assumed knowledge or elsewhere in docs, like at the link above that you could point to, but I think it'd be better to try to call out or draw attention to what's actually different for serverless.

LMK if that makes sense. @markusthoemmes any thoughts? I think you mentioned something similar in another comment re calling out the important fields in sample yaml?

File: serverless/installing-service-mesh-control-plane.adoc [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+       enabled: true
+       labels:
+         app: cluster-local-gateway
+         istio: cluster-local-gateway
+       ports:
+         - name: status-port
+           port: 15020
+         - name: http2
+           port: 80
+           targetPort: 80
```
Comment (markusthoemmes) [inline serverless/installing-service-mesh-control-plane.adoc]: This should now say `targetPort: 8080` (changed in the GDoc)

File: serverless/installing-serverless-operator.adoc [hunk 21]
```diff
@@ -0,0 +1,10 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+= Installing the OpenShift Serverless Operator
+
+The OpenShift Serverless Operator can be installed by using the operator indexed in the OpenShift OperatorHub.
```
Comment (abrennan89) [inline serverless/installing-serverless-operator.adoc]: s/installed by using the operator indexed in the OpenShift OperatorHub. /can be installed using the {product-title} instructions for installing Operators.

Although I would recommend adding this to the assembly instead so that you can just xref to the whole operators assembly rather than have to inc. modules. The CLI commands for example for installing Operators are not in the module below.

You can see what I mean in the KubeFed docs where I've used this: https://github.com/openshift/openshift-docs/blob/master/kubefed/installing-kubefed.adoc

File: serverless/installing-service-mesh-control-plane.adoc [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+* An account with cluster administrator access.
+* The ServiceMesh operator must be installed. 
+
+
+.Procedure
+. Copy the sample YAML file into a `smcp.yaml` file.
+. Apply the YAML file using the command:
++
+----
+ oc apply -f smcp.yaml
```
Comment (abrennan89) [inline serverless/installing-service-mesh-control-plane.adoc]: $ oc apply....

**Review by abrennan89 — CHANGES_REQUESTED** [review]
Some additional feedback

File: modules/serverless-installing-service-mesh-member-roll.adoc [hunk 23]
```diff
@@ -0,0 +1,45 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id="installing-service-mesh-member-roll_{context}"]
+= Installing a ServiceMeshMemberRoll
+
+If Service Mesh is configured for multi-tenancy, you should have a Service Mesh Member Roll for the control plane namespace. For applications to make use of the deployed control plane and ingress, their namespaces need to be part of a member roll.
```
Comment (alanfx) [inline modules/serverless-installing-service-mesh-member-roll.adoc]: AFAIK, I'm not sure you can configure Service Mesh without multi-tenancy any more? I guess we'd have to verify this with the Istio team, but that is my understanding.

Comment (joaedwar) [reply modules/serverless-installing-service-mesh-member-roll.adoc]: @alanfx Should the 'If" be removed and this made into an imperative ( i.e You must have a Service Mesh Member Roll...)

File: serverless/installing-openshift-serverless.adoc [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+:context: installing-openshift-serverless
+
+toc::[]
+
+include:serverless/cluster-sizing-requirements.adoc[leveloffset=+1]
+include:serverless/installing-service-mesh.adoc[leveloffset=+1]
+include:serverless/installing-service-mesh-control-plane.adoc[leveloffset=+1]
+include:modules/serverless-installing-service-mesh-member-roll.adoc[leveloffset=+1]
+include:modules/serverless-installing-knative-serving.adoc[leveloffset+=1]
+include:serverless/installing-serverless-operator.adoc[leveloffset+=1]
```
Comment (adellape) [inline serverless/installing-openshift-serverless.adoc]: All of these `include:`s should have 2 colons: `include::`.

Comment (adellape) [reply serverless/installing-openshift-serverless.adoc]: Same for all other instances of `include:` in other files in this PR.

File: serverless/installing-service-mesh-control-plane.adoc [hunk 25]
```diff
@@ -0,0 +1,97 @@
...
+//
+// serverless/installing-openshift-serverless.adoc
+
+= Installing the ServiceMeshControlPlane
+
+Service Mesh is comprised of a data plane and a control plane. After you install the ServiceMesh operator, you can install the control plane. The control plane manages and configures the sidecar proxies to enforce policies and collect telemetry. The following procedure installs a version of the ServiceMesh control plane that acts as an ingress to your applications.
+
+====
+[NOTE] 
+You need to install the control plane into the `istio-system` namespace. Other namespaces are currently not supported.
```
Comment (adellape) [inline serverless/installing-service-mesh-control-plane.adoc]: s/need to/must/

File: modules/serverless-installing-knative-serving.adoc [hunk 26]
```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id="installing-knative-serving_{context}"]
+= Installing Knative Serving
+
+You need to create a `KnativeServing` object to install Knative Serving using the Openshift Serverless Operator.
```
Comment (adellape) [inline modules/serverless-installing-knative-serving.adoc]: There are a lot of instances throughout the PR of `Openshift` (lowercase `s`). They should all be `OpenShift`.

File: serverless/cluster-sizing-requirements.adoc [hunk 27]
```diff
@@ -0,0 +1,16 @@
...
+= Cluster size requirements
+
+The cluster must be sized appropriately to ensure the instructions work properly. To do so, you can use the MachineSet API to manually scale your cluster up to the desired size.
+
+An Openshift cluster with 10 CPUs and 40 GB memory is the minimum requirement for getting started with your first serverless application. This usually means you will have to scale up one of the default MachineSets by two additional machines.
+
+For more advanced use-cases, such as using Openshift Logging, Monitoring, Metering and Tracing, you will need to deploy more resources. Recommended requirements for such use-cases are 24 vCPUs and 96GB of memory.
+
+// Add or remove an instance of a machine in a MachineSet
+include:modules/machineset-manually-scaling.adoc[leveloffset=+1]
```
Comment (adellape) [inline serverless/cluster-sizing-requirements.adoc]: This module is missing the `.Procedure` after the `.Prerequisites` list.

File: serverless/installing-service-mesh.adoc [hunk 28]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id=installing-service-mesh-operator_{context}]
+= Installing the ServiceMesh operator
+
+Openshift Serverless requires an installed version of ServiceMesh.
+
+include:modules/ossm-operator-install.adoc[leveloffset=+1]
```
Comment (adellape) [inline serverless/installing-service-mesh.adoc]: The `{ProductShortName}` in this module doesn't seem to be rendering properly.

image

Comment (adellape) [reply serverless/installing-service-mesh.adoc]: Same for a number of `{ProductName}` instances.

File: serverless/installing-service-mesh.adoc [hunk 29]
```diff
@@ -0,0 +1,17 @@
...
+
+====
+[NOTE] 
+ServiceMesh documentation requests the installation of Kiali and Jaeger Operators. Kiali and Jaeger operators do not need to be installed unless you plan to use them. They are disabled in the default settings proposed below.
+====
```
Comment (adellape) [inline serverless/installing-service-mesh.adoc]: Should be in this form:

```
[NOTE]
====
ServiceMesh documentation requests the installation of Kiali and Jaeger Operators. Kiali and Jaeger operators do not need to be installed unless you plan to use them. They are disabled in the default settings proposed below.
====
```

Note that there also appears to be a whitespace after `[NOTE] ` above; should be removed.

Comment (adellape) [reply serverless/installing-service-mesh.adoc]: There's a few other instances of this out-of-order `[NOTE`] and `====` blocks.

File: serverless/installing-service-mesh-control-plane.adoc [hunk 30]
```diff
@@ -0,0 +1,97 @@
...
+. Copy the sample YAML file into a `smcp.yaml` file.
+. Apply the YAML file using the command:
++
+----
+ oc apply -f smcp.yaml
+----
+
+
+// Verifying the control plane installation
+include:modules/ossm-control-plane-verify.adoc[leveloffset=+1]
```
Comment (adellape) [inline serverless/installing-service-mesh-control-plane.adoc]: This module has this spot in it:

```
.Procedure

[NOTE]
====
The name of the resource is `istio-installation`.
====
```

This renders funny:

image

Suggest moving the note admonition either before the `.Procedure` or elsewhere in a step.

File: serverless/installing-serverless-operator.adoc (hunk 21 above)
Comment (adellape) [reply serverless/installing-serverless-operator.adoc]: :+1:

**Review by adellape — CHANGES_REQUESTED** [review]
After the `include::`s are fixed (see inline comments), we can get a preview doc build going for everyone to review.

File: serverless/cluster-sizing-requirements.adoc [hunk 31]
```diff
@@ -0,0 +1,23 @@
...
+
+[id="cluster-size-requirements_context"]
+= Cluster size requirements
+include::modules/common-attributes.adoc[]
+:context: cluster-size-requirements
+
+The cluster must be sized appropriately to ensure that OpenShift Serverless can run correctly. You can use the MachineSet API to manually scale your cluster up to the desired size.
+
+An OpenShift cluster with 10 CPUs and 40 GB memory is the minimum requirement for getting started with your first serverless application. This usually means you will have to scale up one of the default MachineSets by two additional machines.
+
```
Comment (mgencur) [inline serverless/cluster-sizing-requirements.adoc]: I think this section is now missing some parts of the updated installation document:

Note: The requirements are greatly depending on the deployed applications. By default, each pod requests ~400m of CPU and the recommendation is based on that. With the given recommendation, such an application can scale up to 10 replicas. Lowering the actual CPU request of the application further pushes that boundary.

For more advanced full-fledged use-cases (for example making use of Monitoring, Metering and Tracing) you’ll need to deploy more resources according to the respective documentation. A good starting point are 24 vCPUs and 96GB of memory.

For OpenShift Logging, there are specific CPU and memory requirements. The Elasticsearch node needs 16G of memory for both memory requests and CPU limits. You might need to add additional nodes to the OpenShift Container Platform cluster to run with the recommended or higher memory. When running the OpenShift Logging on a cluster in AWS the recommended instance type is m4.2xlarge (8 vCPUs, 32 GiB Mem).

File: serverless/installing-service-mesh-control-plane.adoc [hunk 32]
```diff
@@ -0,0 +1,96 @@
...
+   grafana:
+     enabled: false
+   sidecarInjectorWebhook:
+     enabled: false
+----
+<1> Autoscaling is disabled in this version. This release is not intended for production use.
+
+[NOTE]
+====
+Running ServiceMesh with a sidecar injection enabled with OpenShift Serverless is currently not recommended.
```
Comment (mgencur) [inline serverless/installing-service-mesh-control-plane.adoc]: Just a note: The service mesh team has a fix for the blocker issue that prevented us from running with sidecar injection. We might need to revisit what we want to use when the fix lands and testing shows that it works.

**Comment by mgencur** [comment]
Looks good to me now.
QE Approved

File: modules/serverless-installing-knative-serving.adoc [hunk 33]
```diff
@@ -0,0 +1,50 @@
...
+. Copy the sample YAML file into `serving.yaml` and apply it using:
++
+----
+$ oc apply -f serving.yaml
+----
++
+. Verify the installation is complete by using the command:
++
+----
+$ oc get knativeserving/knative-serving -n knative-serving --template='{{range .status.conditions}}{{printf "%s=%s\n" .type .status}}{{end}'
```
Comment (mgencur) [inline modules/serverless-installing-knative-serving.adoc]: This command is missing one curly bracket at the end. Should be ```oc get knativeserving/knative-serving -n knative-serving --template='{{range .status.conditions}}{{printf "%s=%s\n" .type .status}}{{end}}'```

Comment (joaedwar) [reply modules/serverless-installing-knative-serving.adoc]: Got it !! Thanks Martin

**Comment by mgencur** [comment]
QE Approved

**Review by bbrowning — CHANGES_REQUESTED** [review]
Lots of great stuff here - thanks! I've left a few comments with some screenshots of the rendered preview to see where some things are bleeding into the rendered page that shouldn't be visible.

More generally, regarding Service Mesh, it appears all our install docs there are pointing to upstream Maistra/Jaeger/Kiali bits and not the actual productized Service Mesh operators. Also, we reference Maistra TP12 but for our Serverless TP1 release this will need to be the Service Mesh GA release.

We also explicitly disable Kiali, Jaeger, and sidecar injection in our `ServiceMeshControlPlane` example. But the docs walk the user through installing the Jaeger and Kiali operators, which makes me think we should delete that part. Perhaps that's where the unused `modules/serverless-installing-service-mesh-no-kiali-jaeger.adoc` should be included here instead of the full Service Mesh install docs?

**Comment by joaedwar** [comment]
> Lots of great stuff here - thanks! I've left a few comments with some screenshots of the rendered preview to see where some things are bleeding into the rendered page that shouldn't be visible.
> 
> More generally, regarding Service Mesh, it appears all our install docs there are pointing to upstream Maistra/Jaeger/Kiali bits and not the actual productized Service Mesh operators. Also, we reference Maistra TP12 but for our Serverless TP1 release this will need to be the Service Mesh GA release.
> 
> We also explicitly disable Kiali, Jaeger, and sidecar injection in our `ServiceMeshControlPlane` example. But the docs walk the user through installing the Jaeger and Kiali operators, which makes me think we should delete that part. Perhaps that's where the unused `modules/serverless-installing-service-mesh-no-kiali-jaeger.adoc` should be included here instead of the full Service Mesh install docs?

@bbrowning you're absolutely right the modules/serverless-installing-service-mesh-no-kiali-jaeger.adoc should be included to point the user away from installing Kiali, Jeager, and sidecar injection

**Review by adellape — APPROVED** [review]
Structural changes look good! LGTM, just a couple minor comments in-line.
